public/js/mainCtrl.js

```diff
@@ -0,0 +1,41 @@
+alwaysDo.controller('MainCtrl', function() {
+
+  self = this;
+
+  self.title= 'Always Do!';
+  self.task = '';
+  self.taskList = [];
+  self.filter = {};
+  self.taskCount = self.taskList.length;
+
+  self.addTask = function() {
+    self.taskList.push({ 'task' : self.task, 'completed' : false });
+    self.taskCount = self.taskList.length;
+    self.task = '';
+  };
+
+  self.toggleComplete = function(task_index) {
+    self.taskList[task_index].completed = !self.taskList[task_index].completed;
```

Comment:
`$index` can be used in Angular - it's a thing that gets the index of viewmodels